### PR TITLE
Added IBusContext to bus processing context for operators

### PR DESCRIPTION
**Pull request by bullmastiffo**
To resove https://github.com/SDVentures/Contour/issues/21 added IBusContext to bus processing context for operators

### CODE REVIEW

File: Tests/Contour.Common.Tests/PatternsSpecs.cs [hunk 1]
```diff
@@ -287,7 +287,7 @@ public void should_accept_message()
                     "Body");
 
                 Mock<IDelivery> deliveryMock = new Mock<IDelivery>();
-                BusProcessingContext.Current = new BusProcessingContext(deliveryMock.Object);
+                BusProcessingContext.Current = new BusProcessingContext(deliveryMock.Object, null);
```
Comment (semion-akimtsev) [inline Tests/Contour.Common.Tests/PatternsSpecs.cs]: Looks like we should provide a mock here and enforce bus parameter validation inside Bus Processing Context as null does not make any sense for the client code. In other words how was this delivery delivered? The answer imho should be  "by bus" :)

Comment (bullmastiffo) [reply Tests/Contour.Common.Tests/PatternsSpecs.cs]: As long as instances under test don't need it I see no reason to add mocks as long as null suffice.

Comment (semion-akimtsev) [reply Tests/Contour.Common.Tests/PatternsSpecs.cs]: Bus processing context acts as an aggregate for delivery and bus, thus it should maintain it's consistency, e.x. delivery and bus context should both be null or not null.
In other way why should one provide the delivery and not the bus which is related to the delivery?

This is one more reason to push just the consuming context to the constructor and extract delivery and bus inside (assuming a potentially non-type safe cast to IDeliveryContext which should be fixed aside this task).

**Comment by semion-akimtsev** [comment]
@bullmastiffo ,
need to update version info in RELEASE_NOTES.md to 1.4.0 with comments

**Pull request merged**